app/src/test/java/com/example/parkingproyect/ParkingReservationTest.java

```diff
@@ -0,0 +1,226 @@
+    }
+
+    @Test
+    public void validate_parkingNumber_less_parkingSize() {
+        when(model.getParking()).thenReturn(new Parking(14));
+        Assert.assertFalse(presenter.validateParkingLot(15));
+        verify(view).showErrorParkingNumberOutOfRange();
+    }
+    //endregion
+
+    //region reservationDates validators test
+    @Test
+    public void validate_parkingDates_success() {
+        Calendar calendar = Calendar.getInstance();
+        calendar.add(Calendar.DAY_OF_YEAR, 1);
+        Date startDate = calendar.getTime();
+        calendar.add(Calendar.DAY_OF_YEAR, 1);
+        Date endDate = calendar.getTime();
+
+        Assert.assertTrue(presenter.validateReservationDates(startDate.getTime(), endDate.getTime()));
+    }
+
+    @Test
+    public void validate_parkingDates_startDate_and_endDate_empty() {
+        Assert.assertFalse(presenter.validateReservationDates(0, 0));
+        verify(view).showErrorReservationDatesEmpty();
+    }
+
+    @Test
+    public void validate_parkingDates_startDate_empty() {
+        Calendar calendar = Calendar.getInstance();
+        calendar.add(Calendar.DAY_OF_YEAR, 1);
+        Date endDate = calendar.getTime();
+
+        Assert.assertFalse(presenter.validateReservationDates(0, endDate.getTime()));
+        verify(view).showErrorReservationDatesEmpty();
+    }
+
+
+    @Test
+    public void validate_parkingDates_endDate_empty() {
+        Calendar calendar = Calendar.getInstance();
+        calendar.add(Calendar.DAY_OF_YEAR, 1);
+        Date startDate = calendar.getTime();
+
+        Assert.assertFalse(presenter.validateReservationDates(startDate.getTime(), 0));
+        verify(view).showErrorReservationDatesEmpty();
+    }
+
+
+    @Test
+    public void validate_parkingDates_startDate_less_today() {
+        Calendar calendar = Calendar.getInstance();
+        calendar.add(Calendar.DAY_OF_YEAR, -2);
+        Date startDate = calendar.getTime();
+        calendar.add(Calendar.DAY_OF_YEAR, 5);
+        Date endDate = calendar.getTime();
+
+        Assert.assertFalse(presenter.validateReservationDates(startDate.getTime(), endDate.getTime()));
+        verify(view).showErrorDatesOutOfRange();
+    }
+
+    @Test
+    public void validate_parkingDates_endDate_less_startDate() {
+        Calendar calendar = Calendar.getInstance();
+        calendar.add(Calendar.DAY_OF_YEAR, 1);
+        Date startDate = calendar.getTime();
+        calendar.add(Calendar.DAY_OF_YEAR, -2);
+        Date endDate = calendar.getTime();
+
+        Assert.assertFalse(presenter.validateReservationDates(startDate.getTime(), endDate.getTime()));
+        verify(view).showErrorDatesOutOfRange();
+    }
+    //endregion
+
+    @Test
+    public void validate_get_parking_with_list_of_reservations() {
+        when(model.getParking()).thenReturn(new Parking(20));
+        Assert.assertEquals(new Parking(20).getParkingSize(), presenter.getParkingWithReservations().getParkingSize());
```

Comment:
And here do `assertEquals(parkingSize, presenter.getParkingWithReservations().getParkingSize());`